.NET workflows (including `test-codegen`) incompatible with `ubuntu-24.04` runner (now `ubuntu-latest`)

### 👓 What did you see?

Deployment of `ubuntu-latest` runners to `ubuntu-24.04` is currently in progress (5 Dec - 17 Jan) - https://github.com/actions/runner-images/issues/10636.

As of 3 Jan, 2025, 10:33 AM GMT (see [#3808](https://github.com/cucumber/gherkin/actions/runs/12596383788) on https://github.com/cucumber/gherkin/pull/343), `ubuntu-24.04` has become available as `ubuntu-latest` on runners used for this repository and it has been observed that both codegen and dotnet test workflows are now failing.

#### Failing .NET test workflow ([#1710](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560))

<details>
<code>Run dotnet test
  dotnet test
  shell: /usr/bin/bash -e {0}
  env:
    DOTNET_ROOT: /usr/share/dotnet
  Determining projects to restore...
  Restored /home/runner/work/gherkin/gherkin/dotnet/Gherkin/Gherkin.csproj (in 1.0[2](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:2) sec).
  Restored /home/runner/work/gherkin/gherkin/dotnet/Gherkin.SourceGenerator/Gherkin.SourceGenerator.csproj (in 2.04 sec).
  Restored /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/Gherkin.Specs.csproj (in [3](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:3).28 sec).
  Restored /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/Gherkin.Specs.csproj (in [4](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:4).08 sec).
  Restored /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Benchmarks/Gherkin.Benchmarks.csproj (in 4.31 sec).
  Gherkin.SourceGenerator -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin.SourceGenerator/bin/Debug/netstandard2.0/Gherkin.SourceGenerator.dll
  Gherkin -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin/bin/Debug/netstandard2.0/Gherkin.dll
  Gherkin -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin/bin/Debug/net8.0/Gherkin.dll
  Gherkin.Specs -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/bin/Debug/net472/Gherkin.Specs.exe
  Gherkin.Benchmarks -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Benchmarks/bin/Debug/net481/Gherkin.Benchmarks.exe
  Gherkin.Specs -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/bin/Debug/net8.0/Gherkin.Specs.dll
  The package Gherkin.30.0.4 is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
  Successfully created package '/home/runner/work/gherkin/gherkin/dotnet/Gherkin/bin/Debug/NuGet/Gherkin.30.0.4.nupkg'.
  Gherkin.Benchmarks -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Benchmarks/bin/Debug/net8.0/Gherkin.Benchmarks.dll
  Gherkin.Specs -> /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/bin/Debug/net9.0/Gherkin.Specs.dll
Test run for /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/bin/Debug/net9.0/Gherkin.Specs.dll (.NETCoreApp,Version=v9.0)
Test run for /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/bin/Debug/net472/Gherkin.Specs.exe (.NETFramework,Version=v4.7.2)
Test run for /home/runner/work/gherkin/gherkin/dotnet/Gherkin.Specs/bin/Debug/net8.0/Gherkin.Specs.dll (.NETCoreApp,Version=v8.0)
VSTest version 17.12.0 (x[6](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:7)4)
VSTest version 17.12.0 (x64)


VSTest version 1[7](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:8).12.0 (x64)

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
System.IO.FileNotFoundException: Could not find 'mono' host. Make sure that 'mono' is installed on the machine and is available in PATH environment variable.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Helpers.DotnetHostHelper.GetMonoPath() in /_/src/Microsoft.TestPlatform.CoreUtilities/Helpers/DotnetHostHelper.cs:line 93
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager.GetTestHostProcessStartInfo(IEnumerable`1 sources, IDictionary`2 environmentVariables, TestRunnerConnectionInfo connectionInfo) in /_/src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs:line 219
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, String runSettings) in /_/src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs:line 229
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.InitializeTestRun(TestRunCriteria testRunCriteria, IInternalTestRunEventsHandler eventHandler) in /_/src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManager.cs:line 1[8](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:9)4
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Test Run Aborted.

Passed!  - Failed:     0, Passed:   208, Skipped:     0, Total:   208, Duration: 3 s - Gherkin.Specs.dll (net8.0)

Passed!  - Failed:     0, Passed:   208, Skipped:     0, Total:   208, Duration: 3 s - Gherkin.Specs.dll (net[9](https://github.com/cucumber/gherkin/actions/runs/12602459571/job/35125575560#step:4:10).0)
Error: Process completed with exit code 1.
</code>
</details>

#### Failing codegen test workflow ([#3812](https://github.com/cucumber/gherkin/actions/runs/12605153154/job/35133353530))

<details>
<code>Run dotnet tool update Berp --version 1.3.0 --tool-path ~/bin
  dotnet tool update Berp --version 1.3.0 --tool-path ~/bin
  echo "~/bin" >> $GITHUB_PATH
  shell: /usr/bin/bash -e {0}
  env:
    DOTNET_ROOT: /usr/share/dotnet
No usable version of libssl was found
/home/runner/work/_temp/86edf571-0eab-43df-a181-488f7aeb433f.sh: line 1:  [2](https://github.com/cucumber/gherkin/actions/runs/12605153154/job/35133353530#step:4:2)257 Aborted                 (core dumped) dotnet tool update Berp --version 1.[3](https://github.com/cucumber/gherkin/actions/runs/12605153154/job/35133353530#step:4:3).0 --tool-path ~/bin
Error: Process completed with exit code 13[4](https://github.com/cucumber/gherkin/actions/runs/12605153154/job/35133353530#step:4:4).
</code>
</details>

Workflow updates are required to resolve the incompatibilities.

### ✅ What did you expect to see?

Running pipelines.

### 📦 Which tool/library version are you using?

ubuntu-latest (ubuntu-24.04) GitHub runner image

### 🔬 How could we reproduce it?

Run the .NET or codegen test workflows with `ubuntu-latest`

### 📚 Any additional context?

_No response_

--- a/.github/workflows/test-dotnet.yml
+++ b/.github/workflows/test-dotnet.yml
@@ -20,8 +20,7 @@ on:
 
 jobs:
   test-dotnet:
-    # Failing on `ubuntu-24.04` (https://github.com/cucumber/gherkin/issues/349)
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-dotnet@v4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,9 @@ This document is formatted according to the principles of [Keep A CHANGELOG](htt
 
 ## [Unreleased]
 
+### Changed
+- [.NET] Reduce NuGet size by only targeting .NET Standard 2.0
+
 ### Fixed
 - [.NET] Fix NuGet package generation
 

--- a/dotnet/Gherkin.Specs/Gherkin.Specs.csproj
+++ b/dotnet/Gherkin.Specs/Gherkin.Specs.csproj
@@ -1,7 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   
   <PropertyGroup>
-    <TargetFrameworks>net9.0;net8.0;net472</TargetFrameworks>
+    <TargetFrameworks>net9.0;net8.0</TargetFrameworks>
+    <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net472</TargetFrameworks>
     <OutputType>Exe</OutputType>
     <StartupObject>Gherkin.Specs.CLI.Program</StartupObject>
   </PropertyGroup>

--- a/dotnet/Gherkin/Gherkin.csproj
+++ b/dotnet/Gherkin/Gherkin.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   
   <PropertyGroup>
-    <TargetFrameworks>net8.0;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <NoWarn>1591</NoWarn>
     <Deterministic Condition="'$(Configuration)' == 'Release'">false</Deterministic>

